src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -64,13 +57,38 @@ protected virtual IServiceProvider ServiceProvider
             {
                 if (_serviceProvider == null)
                 {
-                    _serviceProvider = _serviceCollection.BuildServiceProvider();
+                    lock (_syncLock)
+                    {
+                        if (_serviceProvider == null)
+                        {
+                            _serviceProvider = _serviceCollection.BuildServiceProvider();
+                        }
+                    }
                 }
 
                 return _serviceProvider;
```

Comment:
I'm thinking maybe this part should also be synced because while this method is called and `_serviceProvider` is not null, it could be refreshed (by nulling it) in other methods.
Gonna update.